core/src/main/java/org/infinispan/statetransfer/StateTransferInterceptor.java

```diff
@@ -357,7 +361,7 @@ private Object handleTxWriteReturn(InvocationContext rCtx, VisitableCommand rCom
          throws Throwable {
       int retryTopologyId = -1;
       WriteCommand writeCommand = (WriteCommand) rCommand;
-      if (t instanceof OutdatedTopologyException) {
+      if (t instanceof OutdatedTopologyException || t instanceof AllOwnersLostException) {
```

Comment:
I was thinking of doing it in the remote get `ResponseFilter`, but you're probably right, the current filter interface isn't a good fit.